pkg/controller/bootstrap/tokencleaner_test.go

```diff
@@ -100,3 +100,31 @@ func TestCleanerNotExpired(t *testing.T) {
 
 	verifyActions(t, expected, cl.Actions())
 }
+
+func TestCleanerExpiredAt(t *testing.T) {
+	cleaner, cl, secrets, err := newTokenCleaner()
+	if err != nil {
+		t.Fatalf("error creating TokenCleaner: %v", err)
+	}
+
+	secret := newTokenSecret("tokenID", "tokenSecret")
+	addSecretExpiration(secret, timeString(3*time.Second))
+	expected := []core.Action{}
+	verifyFunc := func() {
+		secrets.Informer().GetIndexer().Add(secret)
+		cleaner.evalSecret(secret)
+		verifyActions(t, expected, cl.Actions())
+	}
+	// token has not expired currently
+	verifyFunc()
+
+	expected = []core.Action{
+		core.NewDeleteAction(
+			schema.GroupVersionResource{Version: "v1", Resource: "secrets"},
+			api.NamespaceSystem,
+			secret.ObjectMeta.Name),
+	}
+
+	// token expires after 3 seconds
+	time.AfterFunc(3*time.Second, verifyFunc)
```

Comment:
this doesn't test that the secret got auto-added back into the queue. we need to verify the length of the queue is 0, then after the secret expiration, the secret is auto-added back into the queue

Reply:
you are right, I modified the unit test and found the problem with the previous code. GetExpiration I should be trying to return how long it will expire, the previous code returned a negative number